### ISSUE TITLE
Search Query doesn't work on dynamically added options, using an Event Handler on the search field

### ISSUE THREAD

**Issue by venkatdev22** [pytesNET/tail.select#87]
After adding new options through ajax response Items are getting added to existing options. But, search seems to be not working. Below is my select configuration and code sample

//Code sample

document.addEventListener("DOMContentLoaded", function () {
  
    var ulmLastMultiple = tail.select(".ulmLastMultiple", {
        search: true, multiSelectAll: true, placeholder: "Select options"
    });

    //Add key up event lister to location type serarch input
    var locationSerachElemts = document.querySelectorAll(".search-input");
    locationSerachElemts[locationSerachElemts.length - 1].addEventListener('keyup', function (event) {

        var dataName = this.value;
        //console.log("keyup event dataName", dataName);
        $.ajax({
            type: "Get",
            url: '/controller/action?DataName=' + dataName,
            contentType: "application/json; charset=utf-8",
            dataType: "json",
            data: { DataName: dataName },
            success: function (data) {
                var newAddItem = [];
                data.forEach(function (item) {
                                   newAddItem.push({ key: item.value, value: item.name, description: "" })
                });

                //console.log('newAddItem', newAddItem);
                ulmLastMultiple.options.add(newAddItem);
            }
        })

    });
});

_Originally posted by @venkatdev22 in https://github.com/pytesNET/tail.select/issues/13#issuecomment-514178246_

**Comment by SamBrishes** [comment]
Hellow,

first of all: Thanks for your issue. It shows me, that the search system itself is currently being made poorly accessible, especially on such really important purposes and use cases. And it also showed me a bug in the `.config()` method of the main select handler ;D

The first problem is: You "overwrite" and "confuse" the `.query()` method, which builds up the dropdown list, because you hook yourself into the search field. So theoretically you just need to call `ulmLastMultiple.query(this.value)` after adding all available options, but...

the second problem is, that your code adds more and more items, because it calls the ".add()" method everytime the user press' a (new) key into the search field. This doesn't matter on `the tail.select.options` class, but the search method uses the options list of the source `<select>` field, and this field gets filled with your options again and again and again. This results in dozens or hundreds of options if you JUST would call the `.query()` method after adding the items.

This means: A poor programming structure on my side, and a required workaround on your side to get this thing working. It tooked me more then one hour to find the follwing...

## ... Solution
The following example can also be found on [CodePen.io](https://codepen.io/sambrishes/pen/PMNbqV) and works like your example. Single difference: I'm adding the options manually, not using AJAX, but this doesn't really matter in this case.

What happens? Everytime a new "Search Request" has been added, so when the user hits another key, it will clean the source `<select>` field, creates a new `tail.select.options` instance, adds them to the respective main `tail.select` instance and calls a new query using the user value.

```javascript
let select = tail.select("select", {
    search: true, 
    multiSelectAll: true, 
    placeholder: "Select options"
});

select.search.children[0].addEventListener("keyup", function(event){
    
    // Reset 
    [].map.call(select.e.querySelectorAll("[data-select-option='add']"), function(item){
        item.parentElement.removeChild(item);
    });
    [].map.call(select.e.querySelectorAll("[data-select-optgroup='add']"), function(item){
        item.parentElement.removeChild(item);
    });
    
    // Options
    let options = new tail.select.options(select.e, select);
    options.add({
        "option-1": {
            key: "option-1",
            value: "Option First"
        },
        /* more options */
    });
    
    // Add & Query
    select.options = options;
    select.query(this.value);
});
```

Based on your example, it should look like this (untested, of course)

```javascript
var ulmLastMultiple = tail.select(".ulmLastMultiple", {
    search: true, 
    multiSelectAll: true, 
    placeholder: "Select options"
});

ulmLastMultiple.search.children[0].addEventListener("keyup", function(event){
    var dataName = this.value;
    jQuery.ajax({
        type: "Get",
        url: '/controller/action?DataName=' + dataName,
        contentType: "application/json; charset=utf-8",
        dataType: "json",
        data: { DataName: dataName },
        success: function (data) {
            var newAddItem = {};
            data.forEach(function(item){
                newAddItem[key] = { key: item.value, value: item.name, description: "" };
            });
            
            // Reset 
            [].map.call(ulmLastMultiple.e.querySelectorAll("[data-select-option='add']"), function(item){
                item.parentElement.removeChild(item);
            });
            [].map.call(ulmLastMultiple.e.querySelectorAll("[data-select-optgroup='add']"), function(item){
                item.parentElement.removeChild(item);
            });
            
            // Options
            var options = new tail.select.options(ulmLastMultiple.e, ulmLastMultiple);
            options.add(newAddItem);
            
            // Add & Query
            ulmLastMultiple.options = options;
            ulmLastMultiple.query(dataName);
        }
    });
});
```


Sincerely,
Sam.

**Comment by SamBrishes** [comment]
This issue is fixed in the upcoming release 0.6.0
